lib/routes.js.coffee

```diff
@@ -275,30 +275,27 @@ Utils =
 
 
   route_url: (route_defaults) ->
-    return route_defaults if typeof route_defaults == 'string'
+    return route_defaults if typeof route_defaults is 'string'
     protocol = route_defaults.protocol || Utils.current_protocol()
-    hostname = route_defaults.host || window.location.hostname
+    hostname = route_defaults.host || Utils.current_host()
     port = route_defaults.port || (Utils.current_port() unless route_defaults.host)
     port = if port then ":#{port}" else ''
+    "#{protocol}://#{hostname}#{port}"
```

Comment:
it can produce:
`https://null:3000`

Reply:
@krzysiek1507 so if no hosts - no `"#{protocol}://#{hostname}#{port}"` at all? Or can be better solution?

Reply:
@le0pard I was thinking about it and it seems that `hostname` has to be required for `_url` helpers in a `windowless` environment. It doesn't make sense to render HTML with URL like `https://null/products/bag` or `https:///products/bag`. Of course, it makes sense to render path but we want URL. Maybe we should follow Rails' implementation? What do you think?